src/core/dense_set.cc

```diff
@@ -701,7 +716,10 @@ void* DenseSet::AddOrReplaceObj(void* obj, bool has_ttl) {
       dptr = dptr->AsLink();
 
     void* res = dptr->Raw();
-    obj_malloc_used_ -= ObjectAllocSize(res);
+    const size_t res_sz = ObjectAllocSize(res);
+    DCHECK(obj_malloc_used_ >= res_sz) << "Invalid deduction of object size " << res_sz
+                                       << " from total set size " << obj_malloc_used_;
```

Comment:
this prints both values if fails:
```suggestion
    DCHECK_GE(obj_malloc_used_, res_sz);
```

Reply:
changed